track_remote.go

```diff
@@ -0,0 +1,161 @@
+// +build !js
+
+package webrtc
+
+import (
+	"sync"
+
+	"github.com/pion/rtp"
+)
+
+// TrackRemote represents a single inbound source of media
+type TrackRemote struct {
```

Comment:
I think it's better to define an interface for supporting custom remote track.

Reply:
:+1: will do that!

Reply:
Mind if I leave out of this PR? This is a really big change again, I really want to land this weekend. 

I am falling behind on other things :(